### PR TITLE
Fix for Accessibility Bug 586335

### PR DESCRIPTION
**Pull request by lajones**
Fix for Accessibility Bug 586335 - providing key handlers for EntityType shape and Properties & Navigation Properties compartments.

### CODE REVIEW

File: src/EFTools/EntityDesignEntityDesigner/CustomCode/Shapes/EntityTypeShape.cs [hunk 1]
```diff
@@ -690,28 +687,35 @@ public override string AccessibleHelp
...
         }
 
-        public override void OnKeyUp(DiagramKeyEventArgs e)
+        private void EntityTypeShape_KeyUp(object sender, DiagramKeyEventArgs e)
         {
             if (e.Control && e.KeyCode == Keys.Up)
             {
-                var isExpanded = _expandCollapseField.IsExpanded(this);
-                if (isExpanded.HasValue && isExpanded.Value)
+                if (IsExpanded)
```
Comment (divega) [inline src/EFTools/EntityDesignEntityDesigner/CustomCode/Shapes/EntityTypeShape.cs]: Nit: I feel tempted to try to refactor this to have a single extension method, e.g. something like EnsureExpandedState(this NodeShape shape, bool state) that could be reused for compartments and entities. Not sure it is possible, and it is not really necessary, just a matter of reducing code duplication.

Comment (lajones) [reply src/EFTools/EntityDesignEntityDesigner/CustomCode/Shapes/EntityTypeShape.cs]: Agreed. I'll try to find a good place to put it and call it. But if it ends up taking too long I'll just revert to this. I don't want to spend too much time on it, this way is clear to read and it's unlikely we will have to revisit or generalize the collapse/expand code any time in the future.

File: src/EFTools/EntityDesignEntityDesigner/CustomCode/Utils/NodeShapeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                {
+                    using (var txn = nodeShape.Store.TransactionManager.BeginTransaction())
+                    {
+                        nodeShape.IsExpanded = false;
+                        txn.Commit();
+                    }
+                }
+            }
+        }
+    }
```
Comment (divega) [inline src/EFTools/EntityDesignEntityDesigner/CustomCode/Utils/NodeShapeExtensions.cs]: @lajones 
I think you should be able to reduce the whole method to something like:
``` C#
public static void EnsureExpandedState(this NodeShape nodeShape, bool expanded)
{
    if (nodeShape.IsExpanded != expanded)
    {
        using(var txn = nodeShape.Store.TransactionManager.BeginTransaction())
        {
            nodeShape.IsExpanded = expanded;
            txn.Commit();
        }
    }
}
```

**Comment by lajones** [comment]
Checked into master with 282ef81a2

File: src/EFTools/EntityDesignEntityDesigner/CustomCode/Utils/NodeShapeExtensions.cs (hunk 2 above)
Comment (lajones) [reply src/EFTools/EntityDesignEntityDesigner/CustomCode/Utils/NodeShapeExtensions.cs]: Yes, that does look better, but I've already submitted the build and it doesn't make a huge difference - so leaving as it is.

Comment (lajones) [reply src/EFTools/EntityDesignEntityDesigner/CustomCode/Utils/NodeShapeExtensions.cs]: Note: updated in #517

**Pull request merged**